substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/JfrEvent.java

```diff
@@ -32,13 +32,16 @@
 import com.oracle.svm.core.Uninterruptible;
 import com.oracle.svm.core.collections.EnumBitmask;
 import com.oracle.svm.core.thread.JavaThreads;
+import com.oracle.svm.core.VMInspectionOptions;
 
 /**
  * This file contains the VM-level events that Native Image supports on all JDK versions. The event
  * IDs depend on the JDK version (see metadata.xml file) and are computed at image build time.
  */
 public final class JfrEvent {
-    public static final JfrEvent ThreadStart = create("jdk.ThreadStart");
+    private static final int defaultInternalSkipCount = VMInspectionOptions.JfrTrimInternalStackTraces.getValue() ? 3 : 0;
```

Comment:
Can we apply the skip count to all JFR events or are there any events that need to be excluded explicitly?

Reply:
I think it would be fine to apply the skipcount to all events defined in SubstrateVM. The periodic ones and others that don't contain stacktraces would just ignore the skipcount.

Reply:
Ok, then let's use the skipCount for all events. Please remove the static field `defaultInternalSkipCount` and move its logic directly into `getSkipCount()`.